osquery/tables/system/darwin/unified_log.mm

```diff
@@ -0,0 +1,379 @@
+    {GREATER_THAN_OR_EQUALS, NSGreaterThanOrEqualToPredicateOperatorType},
+    {LESS_THAN, NSLessThanPredicateOperatorType},
+    {LESS_THAN_OR_EQUALS, NSLessThanOrEqualToPredicateOperatorType},
+    {LIKE, NSLikePredicateOperatorType}};
+
+const std::map<std::string, std::string> kColumnToOSLogEntryProp = {
+    {"timestamp", "date"},
+    {"message", "composedMessage"},
+    {"storage", "storeCategory"},
+    {"activity", "activityIdentifier"},
+    {"process", "process"},
+    {"pid", "processIdentifier"},
+    {"sender", "sender"},
+    {"tid", "threadIdentifier"},
+    {"subsystem", "subsystem"},
+    {"category", "category"}};
+
+const std::map<std::string, bool> kColumnIsNumeric = {{"timestamp", false},
+                                                      {"message", false},
+                                                      {"storage", true},
+                                                      {"activity", true},
+                                                      {"process", false},
+                                                      {"pid", true},
+                                                      {"sender", false},
+                                                      {"tid", true},
+                                                      {"subsystem", false},
+                                                      {"category", false}};
+
+/**
+ * @brief The backing store keys for saving the lst data extracted.
+ */
+const std::string kUALtimestampKey{"ual_timestamp"};
+const std::string kUALcountKey{"ual_counter"};
+
+/**
+ * @brief maximum rows configuration filter
+ */
+const int kMaxRowsDefault = 100;
+const std::string kMaxRowsColumn = "max_rows";
+const ConstraintOperator kMaxRowsOperator = EQUALS;
+
+/**
+ * @brief timestamp sequential configuration filter
+ */
+const std::string kSentinalColumn = "timestamp";
+const std::string kSentinalValue = "-1";
+const ConstraintOperator kSentinalOperator = GREATER_THAN;
+
+/**
+ * @brief defines the structure that saves the current status for extracting
+ *        sequential data in multiple queries
+ */
+struct SequentialContext {
+  double timestamp; /**< timestamp of last log, used as a pointer */
+  int count; /**< if two or more logs has the same timestamp,
+                  the API will return a pointer to the first one,
+                  save how many we extracted so we can skip to the
+                  current*/
+
+  SequentialContext() : timestamp(0), count(0) {}
+
+  /**
+   * @brief load values from database in case there are stored
+   */
+  void load();
+
+  /**
+   * @brief saves values into database
+   */
+  void save();
+};
+
+void SequentialContext::load() {
+  std::string str;
+  auto s = getDatabaseValue(kPersistentSettings, kUALtimestampKey, str);
+  if (s.ok())
+    timestamp = std::stod(str);
+  s = getDatabaseValue(kPersistentSettings, kUALcountKey, str);
+  if (s.ok())
+    count = std::stod(str);
```

Comment:
`count` is an int. Shouldn't be `std::stod`, right?

Reply:
Yes, should use `std::stoi` instead